modules/foreman_auth_source_ldap.py

```diff
@@ -0,0 +1,233 @@
+  host:
+    description: The hostname of the LDAP server
+    required: true
+  port:
+    description: The port number of the LDAP server
+    required: false
+    tyoe: int
+    default: 389
+  account:
+    description: Account name to use when accessing the LDAP server.
+    required: false
+  account_password:
+    description: Account password to use when accessing the LDAP esrver. Required when using `onthefly_register`
+    required: false
+  base_dn:
+    description: The base DN to use when searching.
+    required: false
+  attr_login:
+    description: Attribute containing login ID. Required when using `onthefly_register`
+    required: false
+  attr_firstname:
+    description: Attribute containing first name. Required when using `onthefly_register`
+    required: false
+  attr_lastname:
+    description: Attribute containing last name. Required when using `onthefly_register`
+    required: false
+  attr_mail:
+    description: Attribute containing email address. Required when using `onthefly_register`
+    required: false
+  attr_photo:
+    description: Attribute containing user photo
+    required: false
+  onthefly_register:
+    description: Whether or not to register users on the fly.
+    required: false
+    type: bool
+  usergroup_sync:
+    description: Whether or not to sync external user groups on login
+    required: false
+    type: bool
+  tls:
+    description: Whether or not to use TLS when contacting the LDAP server.
+    required: false
+    type: bool
+  groups_base:
+    description: Base DN where groups reside.
+    required: false
+    type: bool
+  server_type:
+    description: Type of the LDAP server
+    required: false
+    default: posix
+    choices: ["free_ipa", "active_directory", "posix"]
+  ldap_filter:
+    description: Filter to apply to LDAP searches
+    required: false
+  server_url:
+    description: URL of Foreman server
+    required: true
+  username:
+    description: Username accessing Foreman server
+    required: true
+  password:
+    description: Password of user accessing Foreman server
+    required: true
+  validate_certs:
+    aliases: [ verify_ssl ]
+    description: Whether or not to verify the TLS certificates of Foreman server
+    default: true
+    type: bool
+  organizations:
+    description: List of organizations the authentication source should be assigned to
+    type: list
+  locations:
+    description: List of locations the authentication source should be assigned to
+    type: list
+  state:
+    description: State ot the LDAP authentication source
+    default: present
+    choices: ["present", "absent"]
+'''
+
+EXAMPLES = '''
+- name: LDAP Authentication source
+  foreman_auth_source_ldap:
+    name: "Example LDAP"
+    host: "ldap.example.org"
+    server_url: "https://foreman.example.com"
+    locations:
+      - "Uppsala"
+    organizations:
+      - "Sweden"
+    username: "admin"
+    password: "secret"
+    validate_certs: False
+    state: present
```

Comment:
In an effort to show best practice, would you mind if the exampled used TLS?

Reply:
Fixed